### ISSUE TITLE
Regarding the significantly lower metrics for the “car” and “truck” classes in our reproduction

### ISSUE THREAD

**Issue by mmt-coder** [yuan-zm/DGT-ST#4]
Hello, author. I have a question. During my reproduction of the first stage of PCAN, I noticed that the results for the "car" and "truck" classes differ significantly from those reported in the paper—22.4 and 6.6 points lower, respectively. The other categories are quite close. Is this considered normal?

<img width="880" height="556" alt="Image" src="https://github.com/user-attachments/assets/f93deae8-3c6b-442b-bf53-3842983d5ae3" />

**Comment by yuan-zm** [comment]
Hi, thanks for your question. The results you’re seeing are actually not normal.
In fact, the car usually achieves a high mIoU, while the truck may be more challenging.

May I ask if you used the pre-trained model I provided?
Please note that the model used in the self-training stage is different from the one used in PCAN's first stage.
Make sure you are using the correct pre-trained checkpoint for PCAN.

**Comment by mmt-coder** [comment]
Yes, I used the SourceOnly model you provided for stage-1 training.
Although stage-1 produced abnormal results, I nonetheless trained stage-2 on top of it, and stage-2 performed quite normally, yielding results close to those reported in the paper. 
Could these stage-1 anomalies be caused by the instability of adversarial training?

**Comment by yuan-zm** [comment]
In my experience, using different pretrained models usually has little impact on the mIoU, but it can noticeably shift the IoU of each class.

It could be related to the instability of adversarial training. However, after cleaning the repo I was able to reproduce Stage-1 results close to the paper. Please double-check whether you have modified the Stage-1 code or configuration.

Moreover, beam labels are generated via K-means. I suspect the beam labels you obtained may be different from ours.

**Comment by mmt-coder** [comment]
I guess the beam labels obtained are different. Thank you for your answer.
